ios/browser/ui/webui/brave_webui_source.mm

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/ui/webui/brave_webui_source.h"
+
+#include <string_view>
+#include <vector>
+
+#include "base/containers/flat_map.h"
+#include "base/strings/utf_string_conversions.h"
+#include "brave/components/constants/url_constants.h"
+#include "brave/components/webui/webui_resources.h"
+#include "build/build_config.h"
+#include "components/grit/components_resources.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state.h"
+#include "ios/web/public/webui/web_ui_ios_data_source.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/webui/resource_path.h"
+#include "ui/base/webui/web_ui_util.h"
+
+namespace {
+
+void AddLocalizedStringsBulk(
+    web::WebUIIOSDataSource* html_source,
+    const std::vector<webui::LocalizedString>& localized_strings) {
+  for (auto localized_string : localized_strings) {
+    html_source->AddLocalizedString(std::string(localized_string.name),
```

Comment:
you should be able to call this method here instead now:
```
  virtual void AddLocalizedStrings(
      base::span<const webui::LocalizedString> strings) = 0;
```

but for this you maybe need to update `std::vector<webui::LocalizedString>` to be `std::vector<const webui::LocalizedString>`.